### ISSUE TITLE
AutomatedTestWidgetsFlutterBinding substantially slower to process animations than LiveTestWidgetsFlutterBinding

### ISSUE THREAD

**Issue by jmacpherson** [flutter/flutter#45003]
## Steps to Reproduce

1. Given the following two files:

```
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'scratch.dart';

void main() {
  group('Animation test', () {
    Widget _buildWidget() {
      return ScratchApp();
    }

    Widget _buildPage() {
      return MaterialApp(
        home: Material(
          child: MediaQuery(
            data: MediaQueryData(size: const Size(375, 1012)),
            child: _buildWidget(),
          ),
        ),
      );
    }

    testWidgets("Attempt to reproduce", (WidgetTester tester) async {
      var startTime = DateTime.now();
      var lastTimestamp = startTime;
      print("Start: $startTime");
      await tester.pumpWidget(_buildPage());
      await tester.pumpAndSettle();

      var latestTimestamp = DateTime.now();
      print("Elapsed: ${latestTimestamp.difference(lastTimestamp)}");
      lastTimestamp = latestTimestamp;

      await tester.tap(find.byKey(Key("expand_content")));
      await tester.pumpAndSettle();

      latestTimestamp = DateTime.now();
      print("Elapsed: ${latestTimestamp.difference(lastTimestamp)}");
      lastTimestamp = latestTimestamp;

      await tester.tap(find.byKey(Key("expand_content")));
      await tester.pumpAndSettle();

      latestTimestamp = DateTime.now();
      print("Elapsed: ${latestTimestamp.difference(lastTimestamp)}");
      lastTimestamp = latestTimestamp;

      await tester.tap(find.byKey(Key("expand_content")));
      await tester.pumpAndSettle();

      latestTimestamp = DateTime.now();
      print("Elapsed: ${latestTimestamp.difference(lastTimestamp)}");
      lastTimestamp = latestTimestamp;
    });
  });
}
```

```
import 'package:flutter/material.dart';

class ScratchApp extends StatefulWidget {
  @override
  State<StatefulWidget> createState() {
    return ScratchAppState();
  }
}

class ScratchAppState extends State<ScratchApp>
    with SingleTickerProviderStateMixin {
  bool expanded = false;

  void _toggleExpanded() {
    setState(() {
      expanded = !expanded;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      children: <Widget>[
        Container(
          constraints: BoxConstraints(maxHeight: 500),
          child: SingleChildScrollView(
            child: AnimatedSize(
              vsync: this,
              alignment: Alignment.topCenter,
              curve: Curves.easeInOut,
              duration: Duration(milliseconds: 350),
              child: _buildContent(),
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildContent() {
    return Column(
      children: <Widget>[
        InkWell(
          key: Key("expand_content"),
          onTap: _toggleExpanded,
          child: Row(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              Row(
                children: <Widget>[
                  Text("Expand/Collapse"),
                ],
              )
            ],
          ),
        ),
        AnimatedCrossFade(
          duration: Duration(milliseconds: 500),
          crossFadeState:
              expanded ? CrossFadeState.showSecond : CrossFadeState.showFirst,
          firstChild: Container(),
          secondChild: _buildExpandedContent(),
        ),
      ],
    );
  }

  Widget _buildExpandedContent() {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 8),
      child: Container(
        decoration: BoxDecoration(
          borderRadius: BorderRadius.all(Radius.circular(3)),
          border: Border.all(color: Colors.lightBlue),
        ),
        child: Padding(
          padding: const EdgeInsets.all(8.0),
          child: Column(
            children: <Widget>[
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
              SizedBox(height: 25),
              Text("text"),
            ],
          ),
        ),
      ),
    );
  }
}
```
2. Run the test in scratch_test.dart using `flutter run`, note the elapsed times logged by the test to the console.
3. Then run the test using `flutter test` and compare the elapsed times.

**Results:** When run with `flutter test`, an operation that took 0.8... seconds to perform now takes >6 seconds.

While this specific result is not *too* bad, the implications in larger test suites/applications is problematic. For example, running the same sort of evaluation on one of the tests that inspired this issue shows that a similar animation takes >32 seconds to elapsed using `flutter test`, whereas it takes 0.8 seconds using `flutter run`

The time taken for such animations to evaluate appears to scale relative to the overall height of the widget being animated.

```
$ flutter analyze
Analyzing icecrown...                                               No issues found! (ran in 2.5s)
```

```
[✓] Flutter (Channel stable, v1.9.1+hotfix.5, on Mac OS X 10.14.6
    18G1012, locale en-CA)
    • Flutter version 1.9.1+hotfix.5 at
      /Users/jm/.flutterio/flutter
    • Framework revision 1aedbb1835 (4 weeks ago), 2019-10-17
      08:37:27 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK 
    version 28.0.3)
    • Android SDK at /Users/jm/Library/Android/sdk
    • Android NDK location not configured (optional; useful for
      native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.1, Build version 11A1027
    • CocoaPods version 1.6.1

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 41.1.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[✓] Connected device (1 available)
    • iPad Pro (9.7-inch) • 831BB050-0B84-4EAC-AFC7-B576B82228B0 •
      ios • com.apple.CoreSimulator.SimRuntime.iOS-13-1
      (simulator)

• No issues found!
```

**Comment by tomgilder** [comment]
I've been trying to figure out the reasons for this but so far failing; we're seeing widget tests take a long time (30 seconds) to execute.

It seems really counterintuitive that widget tests would take longer to execute than actually running the app 😃 

Anyone know any workarounds to this?

**Comment by markusaksli-nc** [comment]
This seems fixed on the latest master `1.22.0-10.0.pre.269`.
<details>
 <summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 1.22.0-10.0.pre.269, on Microsoft Windows [Version 10.0.19041.508], locale en-US)
    • Flutter version 1.22.0-10.0.pre.269 at C:\Development\flutter_master
    • Framework revision 51fa704642 (2 days ago), 2020-09-19 11:27:04 -0700
    • Engine revision 4b8477d115
    • Dart version 2.10.0 (build 2.10.0-142.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at C:\Users\marku\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.1
    • Java binary at: C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\193.6626763\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.6.5)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.6.30320.27
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.0)
    • Android Studio at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\193.6626763
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] Connected device (6 available)
    • SM G950F (mobile)           • ce12171c51cc001c03 • android-arm64  • Android 9 (API 28)
    • sdk gphone x86 arm (mobile) • emulator-5554      • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)           • windows            • windows-x64    • Microsoft Windows [Version 10.0.19041.508]
    • Web Server (web)            • web-server         • web-javascript • Flutter Tools
    • Chrome (web)                • chrome             • web-javascript • Google Chrome 85.0.4183.102
    • Edge (web)                  • edge               • web-javascript • Microsoft Edge 85.0.564.51

• No issues found!
```
</details>
<details>
 <summary>Logs</summary>

`flutter run`
```
I/flutter ( 2503): Start: 2020-09-21 10:04:14.192442
I/flutter ( 2503): Elapsed: 0:00:00.730603
I/flutter ( 2503): Elapsed: 0:00:00.152392
I/flutter ( 2503): Elapsed: 0:00:00.116005
I/flutter ( 2503): Elapsed: 0:00:00.119436
I/flutter ( 2503): 00:01 +1: All tests passed!
```
`flutter test`
```
00:05 +0: Animation test Attempt to reproduce
Start: 2020-09-21 10:01:59.885142
Elapsed: 0:00:00.344076
Elapsed: 0:00:00.812183
Elapsed: 0:00:00.101022
Elapsed: 0:00:00.692155
00:07 +1: All tests passed!
```
</details>

Closing this as probably fixed. Could everyone who still has this problem please file a new issue with a minimal reproducible code sample the exact description of the steps, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
